src/extension/debug/dart_debug_impl.ts

```diff
@@ -378,6 +392,15 @@ export class DartDebugSession extends DebugSession {
 		});
 	}
 
+	private subscribeToStreams() {
+		this.observatory.on("Isolate", (event: VMEvent) => this.handleIsolateEvent(event));
+		this.observatory.on("Extension", (event: VMEvent) => this.handleExtensionEvent(event));
+		this.observatory.on("Debug", (event: VMEvent) => this.handleDebugEvent(event));
+
+		const serviceStreamName = this.capabilities.serviceStreamIsPublic ? "Service" : "_Service";
```

Comment:
This looks good - good to be able to switch on the service protocol version in general.

I think you'd also be good just subscribing to both streams and ignoring error results.

Perhaps add a todo: to remove the switch at some point?

Reply:
> I think you'd also be good just subscribing to both streams and ignoring error results.

That was my original plan, but making sure that if they both failed we had something in the log to debug (without always writing logs for the one that failed) was messy, so I figured I may as well do it properly (it seems likely we'll need this if changes for breakpoints across multiple debuggers happen anyway).

> Perhaps add a todo: to remove the switch at some point?

Rather than TODOs all around the place I've started periodically reviewing the Capabilities classes (of which there are now quite a few now) and chopping properties for old things out and fixing the related code appropriately. Unless they're adding complexity, I've been trying to keep them around a while though - this one seems pretty trivial (fun fact: most days Dart Code still has *hundreds* of users using SDK v1.24 🙃!)